bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java

```diff
@@ -264,23 +266,18 @@ public BigtableAsyncAdmin(CommonConnection asyncConnection) throws IOException {
   }
 
   @Override
-  public CompletableFuture<List<TableDescriptor>> listTableDescriptors(List<TableName> list) {
-    if (list == null || list.isEmpty()) {
-      return CompletableFuture.completedFuture(null);
+  public CompletableFuture<List<TableDescriptor>> listTableDescriptors(List<TableName> tableNames) {
+    Preconditions.checkNotNull(tableNames, "tableNames is null");
+    if (tableNames.isEmpty()) {
+      return CompletableFuture.completedFuture(Collections.emptyList());
     }
 
     return toCompletableFuture(bigtableTableAdminClient.listTablesAsync())
         .thenApply(
             t ->
-                list.stream()
+                tableNames.stream()
                     .filter(in -> t.contains(in.getNameAsString()))
-                    .map(
-                        m ->
-                            com.google.bigtable.admin.v2.Table.newBuilder()
-                                .setName(bigtableInstanceName.toTableNameStr(m.getNameAsString()))
-                                .build())
-                    .map(Table::fromProto)
-                    .map(tableAdapter2x::adapt)
+                    .map(tbName -> getDescriptor(tbName).join())
```

Comment:
There is a race condition here: the table might be deleted between a list and get operations Make sure to catch the not found error and remove the table from the list

Reply:
Thanks for detailed feedback, I have addressed this. Please take a look.